var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -33,22 +33,30 @@ class Survey(CMakePackage):
     maintainers = ['jgalarowicz']
 
     version('master', branch='master')
+    version('1.0.3', branch='1.0.3', tag='1.0.3')
```

Comment:
Branch should override tag, did you find you needed to set both of them?

Other than that, all the changes here look good.

Reply:
@scheibelp Right now we only have the 1.0.3 branch, but in a few weeks the branch will be removed and a tag will be applied for 1.0.3.  Is that a practical usage of these features?   I would then remove the branch=1.0.3 clause when we add a 1.0.4 branch/tag combo for the next release.   That way we don't have to submit repeated PRs to update the versions.   I can take out the branch=, if this is not a practical usage of the two clauses (branch=, tag=).  Thanks for looking at this.

Reply:
Unfortunately we don't allow the use of one of these as a backup for the other: it will try branch - preferring it to tag- and if that fails it would not retry with the tag.

However, I will say that because the branch and tag are equal in this case (i.e. both are `"1.0.3"`), and because `git` uses `--branch` for both, this would continue to work even when you change your repository; that being said, `tag` will never be used so long as you have defined `branch` here, so it still makes sense to remove `tag`.

Reply:
@scheibelp Thanks!  I will submit a change.

Reply:
@scheibelp We decided that since we are tagging 1.0.3 very soon we would take off the branch and only leave the tag enabled.   Thanks for your help!    Now we understand how the branch=, tag= clauses work.